src/main/java/com/ericsson/gerrit/plugins/eiffel/handlers/StateHandler.java

```diff
@@ -0,0 +1,207 @@
+        }
+    }
+
+
+    /**
+     * returns true or false depending if the StateHandler is disabled or not.
+     *
+     * @return
+     */
+    private boolean isDisabled() {
+        if (pluginDir == null || pluginDir.toString().isEmpty()) {
+            LOGGER.info(DISABLED_RESPONSE);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * This function sets an SCC eiffel event id for a specific project and branch.
+     *
+     * @param project
+     * @param changeId
+     * @param eiffelEvent
+     */
+    public void setLastSourceChangeCreatedEiffelEvent(String project, String changeId, String eiffelEvent) {
+        if (isDisabled()) {
+            return;
+        }
+
+        DataBaseHandler dBHandler;
+        try {
+            String parentPath = buildParentFilePath(project);
+            createParentDirsIfNecessary(parentPath);
+
+            String fileName = String.format("%s.%s", project, FILE_ENDING);
+            dBHandler = new DataBaseHandler(pluginDir, fileName);
+            String oldEvent = dBHandler.getEventID(Table.SCC_TABLE, changeId);
+            if (!oldEvent.isEmpty()) {
+                dBHandler.updateInto(Table.SCC_TABLE, changeId, eiffelEvent);
+                LOGGER.info("Replaced old event id '{}' with new event if '{}', for project '{}', and branch '{}'.",
+                        oldEvent, eiffelEvent, project, changeId);
+            } else {
+                dBHandler.insertInto(Table.SCC_TABLE, changeId, eiffelEvent);
+                LOGGER.info("Saved eiffel event with id '{}', for project '{}', and branch '{}'.", eiffelEvent, project,
+                        changeId);
+            }
+        } catch (Exception e) {
+            LOGGER.error("Error while trying to insert eiffel event id into database: {}\n{}", e.getMessage(), e);
+        }
+
+    }
+
+    /**
+     * This function returns the last set SCC eiffel event id stored based on the
+     * changeId and specific project.
+     *
+     * @param project
+     * @param changeId
+     * @return
+     */
+    public String getLastSourceChangeCreatedEiffelEvent(String project, String changeId) {
+        if (isDisabled()) {
+            return "";
+        }
+
+        File parentDir = new File(buildParentFilePath(project));
+        if (!(parentDir.exists())) {
+            return "";
+        }
+
+        try {
+            String fileName = String.format("%s.%s", project, FILE_ENDING);
+            DataBaseHandler dBHandler = new DataBaseHandler(pluginDir, fileName);
+            String eventId = dBHandler.getEventID(Table.SCC_TABLE, changeId);
+            LOGGER.info("Fetched old event with id '{}', for project '{}', and branch '{}'", eventId, project,
+                    changeId);
+            return eventId;
+        } catch (Exception e) {
+            LOGGER.error("Error while trying to get eiffel event id from database: {}\n{}", e.getMessage(), e);
+            return "";
```

Comment:
Same here, i don't think we should return empty string if we didnt find anything.

Reply:
RuntimeException is thrown instead